python/NOTICE

```diff
@@ -6,3 +6,22 @@ This product includes software developed at
 The Apache Software Foundation (http://www.apache.org/).
 
 --------------------------------------------------------------------------------
+
+This project includes code from Apache Avro from the Apache Software Foundation with
```

Comment:
The update to LICENSE looks good. And now we can remove this because there isn't anything in the Avro NOTICE that needs to be included. ASF projects don't need to add the boilerplate, too.